src/envoy/http/service_control/client_cache.cc

```diff
@@ -387,6 +393,13 @@ void ClientCache::handleCheckResponse(const Status& http_status,
                 "Google Service Control Check is unavailable, and the "
                 "request is denied due to network fail closed, with error: {}",
                 final_status.error_message());
+
+      // If http_status is not ok, the Code::UNAVAILABLE is from http_status.
+      if (!http_status.ok()) {
+        response_info.error = {httpFailStatusToErrorName(http_status),
```

Comment:
prefer to define a function to take http_status as input and return a "error" struct.